test/inductor/test_torchinductor.py

```diff
@@ -14387,6 +14387,21 @@ def fn(repeat, output_size, data):
             "Generated Triton code should use triton_helpers.minimum for clamping",
         )
 
+    @skipIfMPS
+    def test_inner_reduction_detection(self):
+        if self.device == "cpu":
+            self.skipTest("Skip for CPU device")
+
+        x = torch.randn(100000, 1, 256, device=self.device)
```

Comment:
maybe have another test that also `expand`'s the singleton dim?

Reply:
We can not split reduction in that case right now. To properly test, we need make this improvement first: https://github.com/pytorch/pytorch/issues/167893